v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/transforms/AvroDynamicTransform.java

```diff
@@ -90,19 +99,47 @@ public WriteResult expand(PCollection<KafkaRecord<byte[], byte[]>> kafkaRecords)
             .withFailedInsertRetryPolicy(InsertRetryPolicy.retryTransientErrors())
             .withFormatFunction(kv -> kv.getValue())
             .withExtendedErrorInfo();
+    PCollectionTuple genericRecords;
 
-    writeResult =
-        kafkaRecords
-            .apply(
-                "ConvertKafkaRecordsToGenericRecordsWrappedinFailsafeElement",
-                ParDo.of(
+    genericRecords =
+        kafkaRecords.apply(
+            "ConvertKafkaRecordsToGenericRecordsWrappedinFailsafeElement",
+            ParDo.of(
                     new KafkaRecordToGenericRecordFailsafeElementFn(
-                        options.getSchemaRegistryConnectionUrl())))
+                        options.getSchemaRegistryConnectionUrl(), badRecordRouter))
+                .withOutputTags(
+                    SUCESS_GENERIC_RECORDS, TupleTagList.of(BadRecordRouter.BAD_RECORD_TAG)));
+
+    PCollection<BadRecord> failedGenericRecords =
+        genericRecords.get(BadRecordRouter.BAD_RECORD_TAG);
+    for (ErrorHandler<BadRecord, ?> errorHandler : errorHandlers) {
+      errorHandler.addErrorCollection(
+          failedGenericRecords.setCoder(BadRecord.getCoder(kafkaRecords.getPipeline())));
+    }
+    // TODO: Change the name.
+    PCollectionTuple genericRecords1;
```

Comment:
Can this be renamed to something meaningful?

Reply:
Done